### PR TITLE
Fix FileSystemException in windows on plugin delete

### PR DESCRIPTION
**Pull request by hazemkmammu**
Fixes "java.nio.file.FileSystemException: The process cannot access the
file because it is being used by another process." when deleting an
unloaded plugin.

In windows, deleting the plugin jar file fails because it stays open
despite calling UrlClassLoader.close(). Fixed by using
java.net.URLClassLoader.getResourceAsStream(String) instead of
java.net.URLClassLoader.findResources(String) to load extensions.idx.
The java.net.URLClassLoader.getResourceAsStream(String) collects the
input streams returned (for files and jar files) internally and closes
them when java.net.URLClassLoader.close() is called.

### CODE REVIEW

**Comment by hazemkmammu** [comment]
The test failed in the Java 7 environment because I built the dummy plugin JAR with Java 8.

**Comment by decebals** [comment]
@hazemkmammu 
Thanks for your contribution. I will analyse your code and I will come with a resolution.

File: pf4j/src/main/java/org/pf4j/LegacyExtensionFinder.java [hunk 1]
```diff
@@ -80,12 +81,14 @@ public LegacyExtensionFinder(PluginManager pluginManager) {
             log.debug("Reading extensions storage from plugin '{}'", pluginId);
             Set<String> bucket = new HashSet<>();
 
-            try {
-                Enumeration<URL> urls = ((PluginClassLoader) plugin.getPluginClassLoader()).findResources(getExtensionsResource());
-                if (urls.hasMoreElements()) {
-                    collectExtensions(urls, bucket);
-                } else {
+            log.debug("Read '{}'", getExtensionsResource());
+            PluginClassLoader pluginClassLoader = (PluginClassLoader) plugin.getPluginClassLoader();
```
Comment (hazemkmammu) [inline pf4j/src/main/java/org/pf4j/LegacyExtensionFinder.java]: Casting to PluginClassLoader is unnecessary here.

**Comment by hazemkmammu** [comment]
> @hazemkmammu
> Thanks for your contribution. I will analyse your code and I will come with a resolution.

#318 builds a plugin Jar programmatically for the test and I think is better than packing a jar in resources.

**Comment by decebals** [comment]
Replaced.

**Pull request closed without merging**